### PR TITLE
Fix issue #29, Non-JSON response causes unhandled exception

### PR DESCRIPTION
**Pull request by stefanfisk**

### CODE REVIEW

File: test/invalidResponse.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+const assert = require('assert');
+const proxyquire = require('proxyquire');
+
+const Z = require('../');
+const mockHttpClient = require('./helpers').mockHttpClient;
+
+
+describe('requests()', () => {
```
Comment (richardolsson) [inline test/invalidResponse.js:10]: Why `requests()` when there is no(?) function called `requests` anywhere? I would just call this test spec `requests` or something. No big deal though! 👍

**Review by richardolsson — APPROVED** [review]
Looking good! I had one minor comment, but it's not even worth changing. I'm merging this! 🎉 

Let me know if you need me to fast-track a release to npm for this fix, or if we can just let it be for now and include it in a future release.

File: test/invalidResponse.js (hunk 1 above)
Comment (stefanfisk) [reply test/invalidResponse.js:10]: Ah, I have no idea why I named it like that, probably just entered something and then forgot to fix it before committing.

I don't think that this needs a new release if you haven't encountered it until now. Under normal circumstances everything should work, but when I for example broke my dev environment this made it harder to figure out what was going on.

Comment (richardolsson) [reply test/invalidResponse.js:10]: Yeah, that's a very valid point and I'm glad you fixed it. But lets hold off on the release. In the meantime, maybe you can introduce this change locally and let it sit there while you work on other things, just to confirm that it doesn't break anything (which it shouldn't – a `reject()` is a `reject()` is a `reject()`).

**Pull request merged**